src/Symfony/Component/Security/Core/Exception/DecryptionException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class DecryptionException extends \Exception implements EncryptionExceptionInterface
```

Comment:
It reads weird that Decrypt implements Encrypt 🙂

Maybe having separate exceptions is fine?

Reply:
Hm. Im not sure. But this is how I was reasoning: 

`EncryptionExceptionInterface` catches "all exceptions related to encryption". It is rare that you will do different actions if you are getting an exception when decrypting compared to encrypting. 

However, I agree with you that this is strange.. Maybe rename the `DecryptionException` to `EncryptionException` instead?